### PR TITLE
[SYCL] Add tests for kernel properties

### PR DESCRIPTION
**Pull request by steffenlarsen**
Tests for https://github.com/intel/llvm/pull/6941

### CODE REVIEW

File: SYCL/Basic/sub_group_size_prop.cpp [hunk 1]
```diff
@@ -0,0 +1,117 @@
...
+  assert(ReadSubGroupSize == SGSize &&
+         "Failed check for functor and properties.");
+}
+
+int main() {
+  queue Q;
+  std::vector<size_t> SupportedSGSizes =
+      Q.get_device().get_info<info::device::sub_group_sizes>();
+
+  test<8>(Q, SupportedSGSizes);
```
Comment (v-klochkov) [inline SYCL/Basic/sub_group_size_prop.cpp:112]: I think it may be useful to test the sub-group-size=1, it is the common case for ESIMD and rather a corner case for SYCL but still correct case, right?

**Review by v-klochkov — APPROVED** [review]

File: SYCL/Basic/sub_group_size_prop.cpp (hunk 1 above)
Comment (steffenlarsen) [reply SYCL/Basic/sub_group_size_prop.cpp:112]: Sure! If a device says it can't support it we just won't run the test, so any size we can think of we can add no problem.
